### PR TITLE
modify_sign_up

### PR DESCRIPTION
**Pull request by chan227**
Allow users to preview the avatar before confirm to sign up

### CODE REVIEW

**Comment by chan227** [comment]
ready

File: app/helpers/users_helper.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 module UsersHelper
   def gravatar_for user
-    gravatar_id = Digest::MD5::hexdigest user.email.downcase
-    gravatar_url = "#{Settings.gamer_community.links.gravatar}#{gravatar_id}"
-    image_tag gravatar_url, alt: user.name, class: "gravatar"
+    if user.avatar
+      user.avatar.url
+    else
+      "default.png"
+    end
```
Comment (saobv-2223) [inline app/helpers/users_helper.rb]:
```suggestion
    user.avatar ? user.avatar.url : "default.png"
```
and move default.png to settings file and call it here

File: app/javascript/packs/show.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+$(document).ready(function(){
```
Comment (saobv-2223) [inline app/javascript/packs/show.js]:
```suggestion
$(function() {
```

File: app/javascript/packs/show.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    $(".file").change(function(event){
+       var input = $(event.currentTarget);
+       var file = input[0].files[0];
+       var reader = new FileReader();
+       reader.onload = function(e){
+           image_base64 = e.target.result;
+           preview.attr("src", image_base64);
+       };
+       reader.readAsDataURL(file);
+    });
```
Comment (saobv-2223) [inline app/javascript/packs/show.js]: we use ' in js and css files

File: app/views/users/_form.html.erb [hunk 4]
```diff
@@ -1,17 +1,22 @@
 <%= form_for @user do |f| %>
   <%= render "shared/error_messages", object: f.object %>
 
+  <div class="upload-preview">
+    <img width="250" height="250" alt="Your Avatar" class="center">
```
Comment (saobv-2223) [inline app/views/users/_form.html.erb:5]: use I18n for text and move 250 to settings

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -2,10 +2,15 @@
...
-      <h1>
-        <%= gravatar_for @user %>
-        <%= @user.name %>
-      </h1>
+      <div class="col-sm-3">
+        <% if @user.avatar? %>
+          <%= image_tag @user.avatar.url, height: Settings.users.image.height, width: Settings.users.image.width, class: "img-circle" %>
+        <% else %>
+          <%= image_tag "default.png", height: Settings.users.image.height, width: Settings.users.image.width, class: "img-circle" %>
+        <% end %>
```
Comment (saobv-2223) [inline app/views/users/show.html.erb:10]:
```suggestion
        <%= image_tag gravatar_for(@user), height: Settings.users.image.height, width: Settings.users.image.width, class: "img-circle" %>
```

**Review by saobv-2223 — COMMENTED** [review]
Please, don't add image to GITHUB.

File: app/javascript/packs/show.js (hunk 3 above)
Comment (chan227) [reply app/javascript/packs/show.js]: done

File: app/helpers/users_helper.rb (hunk 1 above)
Comment (chan227) [reply app/helpers/users_helper.rb]: done

File: app/javascript/packs/show.js (hunk 2 above)
Comment (chan227) [reply app/javascript/packs/show.js]: done

**Comment by chan227** [comment]
ready

**Comment by ngocvt-0484** [comment]
add public/upload to gitignore please

**Comment by ngocvt-0484** [comment]
commented

**Comment by chan227** [comment]
ready

File: app/views/users/show.html.erb (hunk 5 above)
Comment (saobv-2223) [inline app/views/users/show.html.erb]:
```suggestion
        <%= image_tag gravatar_for(@user), height: Settings.users.image.height, width: Settings.users.image.width, class: "img-circle" %>
```
I see you wrote gravatar_for action, therefore you can call it here

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
ready

**Pull request merged**